File: ChatBot/ChatBot/ChatViewController.m
#import "ChatViewController.h"
#import "SettingsViewController.h"
#import "NetworkManager.h"
#import "MessageCell.h"

@interface ChatViewController () <UITableViewDelegate, UITableViewDataSource, UITextFieldDelegate>

@end

@implementation ChatViewController

- (instancetype)initWithConversation:(Conversation *)conversation {
    self = [super init];
    if (self) {
        self.currentConversation = conversation;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = self.currentConversation.title;
    self.view.backgroundColor = [UIColor systemBackgroundColor];
    
    // 设置导航栏按钮
    UIBarButtonItem *settingsButton = [[UIBarButtonItem alloc] initWithImage:[UIImage systemImageNamed:@"gear"]
                                                                       style:UIBarButtonItemStylePlain
                                                                      target:self
                                                                      action:@selector(openSettings)];
    self.navigationItem.rightBarButtonItem = settingsButton;
    
    // 添加返回按钮
    UIBarButtonItem *backButton = [[UIBarButtonItem alloc] initWithImage:[UIImage systemImageNamed:@"list.bullet"]
                                                                   style:UIBarButtonItemStylePlain
                                                                  target:self
                                                                  action:@selector(backToList)];
    self.navigationItem.leftBarButtonItem = backButton;
    
    // 设置聊天表格视图
    self.chatTableView = [[UITableView alloc] init];
    self.chatTableView.delegate = self;
    self.chatTableView.dataSource = self;
    self.chatTableView.separatorStyle = UITableViewCellSeparatorStyleNone;
    self.chatTableView.backgroundColor = [UIColor systemBackgroundColor];
    [self.chatTableView registerClass:[MessageCell class] forCellReuseIdentifier:@"MessageCell"];
    [self.view addSubview:self.chatTableView];
    
    // 设置输入框和发送按钮
    UIView *inputView = [[UIView alloc] init];
    inputView.backgroundColor = [UIColor systemGray6Color];
    [self.view addSubview:inputView];
    
    self.messageTextField = [[UITextField alloc] init];
    self.messageTextField.placeholder = @"输入消息...";
    self.messageTextField.borderStyle = UITextBorderStyleRoundedRect;
    self.messageTextField.delegate = self;
    [inputView addSubview:self.messageTextField];
    
    self.sendButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.sendButton setTitle:@"发送" forState:UIControlStateNormal];
    [self.sendButton addTarget:self action:@selector(sendMessage) forControlEvents:UIControlEventTouchUpInside];
    [inputView addSubview:self.sendButton];
    
    // 设置约束
    self.chatTableView.translatesAutoresizingMaskIntoConstraints = NO;
    inputView.translatesAutoresizingMaskIntoConstraints = NO;
    self.messageTextField.translatesAutoresizingMaskIntoConstraints = NO;
    self.sendButton.translatesAutoresizingMaskIntoConstraints = NO;
    
    [NSLayoutConstraint activateConstraints:@[
        [self.chatTableView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
        [self.chatTableView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [self.chatTableView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [self.chatTableView.bottomAnchor constraintEqualToAnchor:inputView.topAnchor],
        
        [inputView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [inputView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [inputView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor],
        [inputView.heightAnchor constraintEqualToConstant:60],
        
        [self.messageTextField.leadingAnchor constraintEqualToAnchor:inputView.leadingAnchor constant:16],
        [self.messageTextField.centerYAnchor constraintEqualToAnchor:inputView.centerYAnchor],
        [self.messageTextField.trailingAnchor constraintEqualToAnchor:self.sendButton.leadingAnchor constant:-8],
        
        [self.sendButton.trailingAnchor constraintEqualToAnchor:inputView.trailingAnchor constant:-16],
        [self.sendButton.centerYAnchor constraintEqualToAnchor:inputView.centerYAnchor],
        [self.sendButton.widthAnchor constraintEqualToConstant:60]
    ]];
}

#pragma mark - Actions

- (void)backToList {
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)openSettings {
    SettingsViewController *settingsVC = [[SettingsViewController alloc] init];
    [self.navigationController pushViewController:settingsVC animated:YES];
}

- (void)sendMessage {
    NSString *message = self.messageTextField.text;
    if (message.length > 0) {
        Message *userMessage = [Message messageWithContent:message isUser:YES];
        [self.currentConversation addMessage:userMessage];
        [self.chatTableView reloadData];
        self.messageTextField.text = @"";
        
        // 滚动到底部
        NSIndexPath *lastIndexPath = [NSIndexPath indexPathForRow:self.currentConversation.messages.count - 1 inSection:0];
        [self.chatTableView scrollToRowAtIndexPath:lastIndexPath atScrollPosition:UITableViewScrollPositionBottom animated:YES];
        
        [[NetworkManager sharedManager] sendMessage:message completion:^(NSString * _Nullable response, NSError * _Nullable error) {
            if (error) {
                UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"错误"
                                                                             message:error.localizedDescription
                                                                      preferredStyle:UIAlertControllerStyleAlert];
                UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"确定"
                                                                  style:UIAlertActionStyleDefault
                                                                handler:nil];
                [alert addAction:okAction];
                [self presentViewController:alert animated:YES completion:nil];
                return;
            }
            
            if (response) {
                Message *aiMessage = [Message messageWithContent:response isUser:NO];
                [self.currentConversation addMessage:aiMessage];
                [self.chatTableView reloadData];
                
                // 滚动到底部
                NSIndexPath *lastIndexPath = [NSIndexPath indexPathForRow:self.currentConversation.messages.count - 1 inSection:0];
                [self.chatTableView scrollToRowAtIndexPath:lastIndexPath atScrollPosition:UITableViewScrollPositionBottom animated:YES];
            }
        }];
    }
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.currentConversation.messages.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    MessageCell *cell = [tableView dequeueReusableCellWithIdentifier:@"MessageCell" forIndexPath:indexPath];
    Message *message = self.currentConversation.messages[indexPath.row];
    [cell configureWithMessage:message];
    return cell;
}

#pragma mark - UITextFieldDelegate

- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [self sendMessage];
    return YES;
}

@end 

File: ChatBot/ChatBot/ChatViewController.swift
import UIKit

class ChatViewController: UIViewController {
    private let tableView = UITableView()
    private let messageInputView = UIView()
    private let messageTextField = UITextField()
    private let sendButton = UIButton(type: .system)
    
    var currentConversation: Conversation?
    private var messages: [Message] = []
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        setupKeyboardHandling()
        
        if currentConversation == nil {
            currentConversation = Conversation(title: "新对话 \(Date().formatted())")
        }
        
        messages = currentConversation?.messages ?? []
        tableView.reloadData()
    }
    
    private func setupUI() {
        // 设置导航栏
        title = currentConversation?.title ?? "新对话"
        navigationController?.navigationBar.prefersLargeTitles = false
        
        // 设置表格视图
        tableView.translatesAutoresizingMaskIntoConstraints = false
        tableView.delegate = self
        tableView.dataSource = self
        tableView.register(MessageCell.self, forCellReuseIdentifier: "MessageCell")
        tableView.separatorStyle = .none
        tableView.keyboardDismissMode = .interactive
        tableView.backgroundColor = .systemBackground
        tableView.contentInset = UIEdgeInsets(top: 8, left: 0, bottom: 8, right: 0)
        view.addSubview(tableView)
        
        // 设置输入视图
        messageInputView.translatesAutoresizingMaskIntoConstraints = false
        messageInputView.backgroundColor = .systemBackground
        messageInputView.layer.borderWidth = 0.5
        messageInputView.layer.borderColor = UIColor.systemGray4.cgColor
        view.addSubview(messageInputView)
        
        // 设置输入框
        messageTextField.translatesAutoresizingMaskIntoConstraints = false
        messageTextField.placeholder = "输入消息..."
        messageTextField.borderStyle = .roundedRect
        messageTextField.backgroundColor = .systemGray6
        messageTextField.font = .systemFont(ofSize: 16)
        messageTextField.delegate = self
        messageInputView.addSubview(messageTextField)
        
        // 设置发送按钮
        sendButton.translatesAutoresizingMaskIntoConstraints = false
        sendButton.setTitle("发送", for: .normal)
        sendButton.titleLabel?.font = .systemFont(ofSize: 16, weight: .medium)
        sendButton.addTarget(self, action: #selector(sendMessage), for: .touchUpInside)
        messageInputView.addSubview(sendButton)
        
        // 设置约束
        NSLayoutConstraint.activate([
            // 表格视图约束
            tableView.topAnchor.constraint(equalTo: view.topAnchor),
            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
            tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
            tableView.bottomAnchor.constraint(equalTo: messageInputView.topAnchor),
            
            // 输入视图约束
            messageInputView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
            messageInputView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
            messageInputView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
            messageInputView.heightAnchor.constraint(equalToConstant: 60),
            
            // 输入框约束
            messageTextField.leadingAnchor.constraint(equalTo: messageInputView.leadingAnchor, constant: 16),
            messageTextField.topAnchor.constraint(equalTo: messageInputView.topAnchor, constant: 10),
            messageTextField.bottomAnchor.constraint(equalTo: messageInputView.bottomAnchor, constant: -10),
            messageTextField.trailingAnchor.constraint(equalTo: sendButton.leadingAnchor, constant: -8),
            
            // 发送按钮约束
            sendButton.trailingAnchor.constraint(equalTo: messageInputView.trailingAnchor, constant: -16),
            sendButton.centerYAnchor.constraint(equalTo: messageInputView.centerYAnchor),
            sendButton.widthAnchor.constraint(equalToConstant: 44),
            sendButton.heightAnchor.constraint(equalToConstant: 40)
        ])
    }
    
    private func setupKeyboardHandling() {
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(keyboardWillShow),
            name: UIResponder.keyboardWillShowNotification,
            object: nil
        )
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(keyboardWillHide),
            name: UIResponder.keyboardWillHideNotification,
            object: nil
        )
    }
    
    @objc private func keyboardWillShow(_ notification: Notification) {
        guard let keyboardFrame = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect else { return }
        let keyboardHeight = keyboardFrame.height
        
        UIView.animate(withDuration: 0.3) {
            self.messageInputView.transform = CGAffineTransform(translationX: 0, y: -keyboardHeight)
        }
    }
    
    @objc private func keyboardWillHide(_ notification: Notification) {
        UIView.animate(withDuration: 0.3) {
            self.messageInputView.transform = .identity
        }
    }
    
    @objc private func sendMessage() {
        guard let text = messageTextField.text?.trimmingCharacters(in: .whitespacesAndNewlines),
              !text.isEmpty else { return }
        
        let message = Message(content: text, isUser: true)
        currentConversation?.addMessage(message)
        messages.append(message)
        
        messageTextField.text = ""
        tableView.reloadData()
        scrollToBottom()
        
        // 模拟AI回复
        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
            let aiMessage = Message(content: "这是一个模拟的AI回复消息。", isUser: false)
            self.currentConversation?.addMessage(aiMessage)
            self.messages.append(aiMessage)
            self.tableView.reloadData()
            self.scrollToBottom()
        }
    }
    
    private func scrollToBottom() {
        guard !messages.isEmpty else { return }
        let lastIndex = IndexPath(row: messages.count - 1, section: 0)
        tableView.scrollToRow(at: lastIndex, at: .bottom, animated: true)
    }
}

// MARK: - UITableViewDataSource
extension ChatViewController: UITableViewDataSource {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return messages.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "MessageCell", for: indexPath) as! MessageCell
        let message = messages[indexPath.row]
        cell.configure(with: message)
        return cell
    }
}

// MARK: - UITableViewDelegate
extension ChatViewController: UITableViewDelegate {
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return UITableView.automaticDimension
    }
}

// MARK: - UITextFieldDelegate
extension ChatViewController: UITextFieldDelegate {
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        sendMessage()
        return true
    }
} 